scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -59,6 +61,25 @@ object BeamOp extends Serializable {
   )(implicit ordK: Ordering[K], kryoCoder: KryoCoder): PCollection[KV[K, java.lang.Iterable[U]]] = {
     reduceFn match {
       case ComposedMapGroup(f, g) => planMapGroup(planMapGroup(pcoll, f), g)
+      case EmptyGuard(MapValueStream(SumAll(pqm: PriorityQueueMonoid[V]))) =>
+        pcoll.apply(MapElements.via(
+          new SimpleFunction[KV[K, java.lang.Iterable[V]], KV[K, java.lang.Iterable[U]]]() {
+            override def apply(input: KV[K, lang.Iterable[V]]): KV[K, java.lang.Iterable[U]] = {
+              // We are not using plus method defined in PriorityQueueMonoid as it is mutating
+              // input Priority Queues. We create a new PQ from the individual ones.
+              // We didn't use Top PTransformation in beam as it is not needed, also
+              // we cannot access `max` defined in PQ monoid.
```

Comment:
why not extract the comparator: `val cmp = pqm.zero.comparator()` and then use Top.of as @tlazaro suggested here:

https://github.com/twitter/scalding/issues/1947#issuecomment-923522362

Reply:
Top requires `max` no of elements in its constructor. But we cannot access it from the monoid.

Reply:
ahh... man that PriorityQueueMonoid is really bad. :(

ahh, but there is a way out!

https://github.com/twitter/algebird/blob/0c45f9395020ceecd28899fe156b7e3c92b29814/algebird-core/src/main/scala/com/twitter/algebird/mutable/PriorityQueueMonoid.scala#L27

you could subclass it! `class ScaldingPriorityQueueMonoid[K](val count: Int)(implicit val ordering: Ordering[K]) extends PriorityQueueMonoid(count)(ordering)` and then you can access the count and the ordering.

Reply:
I made this quick PR: https://github.com/twitter/algebird/pull/1008